add expr package

ozym opened the pull request:
This is a helper package that makes adding numerical configuration where it makes more sense to use an equation rather than the final answer.

bpeng commented on internal/expr/expr.go:
```diff
@@ -0,0 +1,106 @@
+	if err != nil {
+		return 0, err
+	}
+	res, _ := constant.Float64Val(val)
```
Does it make sense to validate the type before getting the value, to avoid possible panic?

ozym replied:
I think, to some extent this is already handled by the underlying code.

e.g.
https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/go/constant/value.go;l=548

ozym replied:
but yes, there's a panic in the.

bpeng reviewed (commented):
Looks good, little suggestion which may not be necessary

ozym commented:
@bpeng I have added some guardrails around the values after evaluating, this is a good enhancement as I suspect this may have caused issues when dealing with floating point values that are given as integers.

I've also added a complex number value, as this is something that crops up a lot in response configuration.

bpeng reviewed (approved):
OK
